teps/0096-pipelines-v1-api.md

```diff
@@ -174,11 +182,106 @@ V1 work should prioritize stabilizing existing beta features that support the [u
+| Run         | alpha         | beta           |
+
+#### API Changes
+
+- Fix [pain points](https://github.com/tektoncd/pipeline/issues/3792) associated with TaskRun and PipelineRun Status.
+- Consider replacing any Task or Pipeline fields that should support parameterization with Strings and performing our own validation.
+  - See [Handling parameter interpolation in fields not designed for it](https://github.com/tektoncd/pipeline/issues/1530) for more details.
+
+#### Deprecations
+
+- Deprecate `PipelineRun.Timeout` and [upgrade `PipelineRun.Timeouts` to beta](https://github.com/tektoncd/pipeline/issues/4460).
+- Remove `Pipeline.Tasks[].Conditions`. Conditions have been replaced by when expressions, which are already in beta.
+- Because PipelineResources are deprecated, remove the `Resources` fields of `Task`, `TaskRun`, `Pipeline`, and `PipelineRun`.
+    - [Still under discussion]: Additional community feedback is needed to determine the minimum V1 requirements that will make it usable for PipelineResource users. We should provide PipelineResource users with a migration path to the V1 API, which may include blocking on features proposed in [TEP-0074: Deprecate PipelineResources](https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md).
+    - We will proceed with [TEP-0074: Deprecate PipelineResources](https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md)
+    independently of Pipelines V1, and merging this TEP as implementable isn't a V1 blocker.
+
+#### Behavior flags
+
+Existing feature flags can be found [here](https://github.com/tektoncd/pipeline/blob/main/config/config-feature-flags.yaml).
+The feature flags "disable-home-env-overwrite", "disable-working-directory-overwrite", and "scope-when-expressions-to-task"
+are not present in this table, as their deprecation has already [been announced](https://github.com/tektoncd/pipeline/blob/main/docs/deprecations.md#deprecation-table).
+
+| Flag                                          | Current Default | Proposed State                                       |
+| --------------------------------------------- | --------------- | -----------------------------------------------------|
+| disable-affinity-assistant                    | false           | default to true                                      |
+| disable-creds-init                            | false           | no change                                            |
+| running-in-environment-with-injected-sidecars | true            | no change                                            |
+| require-git-ssh-secret-known-hosts            | false           | no change                                            |
+| enable-tekton-oci-bundles                     | false           | collapsed under enable-api-fields                    |
+| enable-custom-tasks                           | false           | collapsed under enable-api-fields                    |
```

Comment:
Two questions here:
- Did we mark those as deprecated ?
- Do we want to have OCI bundles stable for v1 ? (my take would be **yes**)

Reply:
Great point, I think this was an oversight on my part that we haven't made any communications yet on these feature flags. I wasn't thinking of this as a "deprecation" since the features are still around.

As for stabilizing OCI bundles, @sbwsg can you weigh in here? It sounds like they won't have to change much for remote resolution. I don't have a good sense on how well they are working right now.

Reply:
Ok, it sounds like we don't have a strong sense yet of how OCI bundles and remote resolution will work together. I think since remote resolution isn't a V1 blocker, we should also not make stabilizing OCI bundles a v1 blocker. Thoughts?

(FYI @jerop @pritidesai since we discussed this in our v1 meeting today)

Reply:
Note that there is two things here : 
- Marking the oci bundle "spec" as stable (how to create an oci bundle, …).
- Marking the use of oci bundle as stable.

Reply:
Discussed offline; the [Tekton OCI bundle contract](https://github.com/tektoncd/pipeline/blob/main/docs/tekton-bundle-contracts.md) is a separate API and out of scope for discussions of the Pipelines API. Marking the use of OCI bundle as stable will have to wait for more answers on remote resolution, so I've added it to the PR as out of scope for V1.

In terms of communicating with users about flag changes, is there a certain length of time we need to wait here? My plan is to collapse these flags only in the v1 API (not in v1beta1), and the features aren't going away, so I'm not sure there is a waiting period required by our compatibility policy.

I also updated the flags section to indicate the stability levels required to enable the OCI bundle and custom tasks features.

Reply:
Late follow-up - Here's a first draft of the oci bundles resolver for remote resolution: https://github.com/tektoncd/resolution/tree/main/bundleresolver